# commit 5ee5d347cf90c6a5a2d4242c7cdfdad1c30ebba1
# Author: Ken Lam <kwlam@sierrawireless.com>
# Date:   Mon Jul 16 14:58:22 2018 +0800
#
# sound: mdm9607 without codec fails to sleep after a voice call
#
# Modified-by: Dragan Marinkovic <dmarinkovi@sierrawireless.com>
diff --git a/asoc/mdm9607.c b/asoc/mdm9607.c
index c55eb5a5..417fc15a 100644
--- a/asoc/mdm9607.c
+++ b/asoc/mdm9607.c
@@ -2381,37 +2381,6 @@ static struct snd_soc_dai_link mdm_wm8944_dai[] = {
 	},
 };
 
-static struct snd_soc_dai_link mdm_nocodec_dai[] = {
-	{
-		.name = LPASS_BE_PRI_MI2S_RX,
-		.stream_name = "Primary MI2S Playback",
-		.cpu_dai_name = "msm-dai-q6-mi2s.0",
-		.platform_name = "msm-pcm-routing",
-		.codec_name = "msm-stub-codec.1",
-		.codec_dai_name = "msm-stub-rx",
-		.dpcm_capture = 1,
-		.dpcm_playback = 1,
-		.no_pcm = 1,
-		.id = MSM_BACKEND_DAI_PRI_MI2S_RX,
-		.be_hw_params_fixup = &mdm_mi2s_rx_be_hw_params_fixup,
-		.ops = &mdm_mi2s_be_ops,
-	},
-	{
-		.name = LPASS_BE_PRI_MI2S_TX,
-		.stream_name = "Primary MI2S Capture",
-		.cpu_dai_name = "msm-dai-q6-mi2s.0",
-		.platform_name = "msm-pcm-routing",
-		.codec_name = "msm-stub-codec.1",
-		.codec_dai_name = "msm-stub-tx",
-		.dpcm_capture = 1,
-		.dpcm_playback = 1,
-		.no_pcm = 1,
-		.id = MSM_BACKEND_DAI_PRI_MI2S_TX,
-		.be_hw_params_fixup = &mdm_mi2s_tx_be_hw_params_fixup,
-		.ops = &mdm_mi2s_be_ops,
-	},
-};
-
 static struct snd_soc_dai_link mdm_tomtom_dai_links[
 				ARRAY_SIZE(mdm_dai) +
 				ARRAY_SIZE(mdm_9330_dai)];
@@ -2424,10 +2393,6 @@ static struct snd_soc_dai_link mdm_wm8944_dai_links[
 				ARRAY_SIZE(mdm_dai) +
 				ARRAY_SIZE(mdm_wm8944_dai)];
 
-static struct snd_soc_dai_link mdm_nocodec_dai_links[
-				ARRAY_SIZE(mdm_dai) +
-				ARRAY_SIZE(mdm_nocodec_dai)];
-
 static struct snd_soc_card snd_soc_card_mdm_wm8944 = {
 	.name = "mdm9607-wm8944-i2s-snd-card",
 	.dai_link = mdm_wm8944_dai_links,
@@ -2436,8 +2401,8 @@ static struct snd_soc_card snd_soc_card_mdm_wm8944 = {
 
 static struct snd_soc_card snd_soc_card_mdm_nocodec = {
 	.name = "mdm9607-nocodec-snd-card",
-	.dai_link = mdm_nocodec_dai_links,
-	.num_links = ARRAY_SIZE(mdm_nocodec_dai_links),
+	.dai_link = mdm_dai,
+	.num_links = ARRAY_SIZE(mdm_dai),
 };
 
 static struct snd_soc_card snd_soc_card_mdm_9330 = {
@@ -2683,14 +2648,8 @@ static struct snd_soc_card *populate_snd_card_dailinks(struct device *dev)
 	}
 	else {
 		card = &snd_soc_card_mdm_nocodec;
-		len_1 = ARRAY_SIZE(mdm_dai);
-		len_2 = len_1 + ARRAY_SIZE(mdm_nocodec_dai);
-
-		memcpy(mdm_nocodec_dai_links, mdm_dai,
-			   sizeof(mdm_dai));
-		memcpy(mdm_nocodec_dai_links + len_1, mdm_nocodec_dai,
-			   sizeof(mdm_nocodec_dai));
-		dailink = mdm_nocodec_dai_links;
+		len_2 = ARRAY_SIZE(mdm_dai);
+		dailink = mdm_dai;
 	}
 
 	if (card) {
